static/app/components/replays/replayContext.tsx

```diff
@@ -181,6 +193,14 @@ type Props = {
 
   replay: ReplayReader | null;
 
+  /**
+   * If provided, the replay will be clipped to this window.
+   */
+  clipWindow?: {
+    durationMs: number;
+    startTimeOffsetMs: number;
+  };
```

Comment:
I'm thinking if this should be a property on `ReplayReader`. Idk yet, and it's something we can change later.

`getFramesByColumn()` has a filter call inside it. If we show other things, like a list of breadcrumbs, then we will need to do that same filter() call. It's not a case now, but in the future we'll need to remember and keep these filter things in sync or something. 

But if clipWindow were inside the reader, then we could return only the relevant frames there. We would still need to return all the rrwebFrames up to the end of the duration, so there's still some time munging to do in here.

Reply:
Ooh I think this is a great suggestion. I'll play around with it and see if I can change to using the reader instead.

Reply:
After a good amount of effort towards this I'm throwing in the towel. I couldn't quite get rrweb to be happy with a truncated set of events, but I don't know why. I may circle back to this at some point though because this solution does seem cleaner. Maybe we can pair on it.